### PR TITLE
Use docker for Ubuntu 18.04 packaging

### PR DESCRIPTION
**Pull request by Yaraslaut**

### CODE REVIEW

**Comment by christianparpart** [comment]
For as long as @whisperity needs Ubuntu 18.04 at work, we need to figure out a way to build binaries on Ubuntu 18.04

What could work, is, adding a new action that builds Ubuntu images through docker. That way we can support building binaries that are not directly supported by GitHub CI runners.

**Comment by Yaraslaut** [comment]
@christianparpart  Please squash if no modifications is required to merge

File: .github/ubuntu18/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    software-properties-common \
+    libqt5gui5 \
+    libqt5opengl5-dev \
+    libqt5x11extras5-dev \
+    qtbase5-dev \
+    qtmultimedia5-dev
+
+RUN add-apt-repository ppa:ubuntu-toolchain-r/test
+RUN apt-get update
+RUN apt-get install -y g++-10
```
Comment (christianparpart) [inline .github/ubuntu18/Dockerfile:34]: I think we can drop this line, meaning, a default compiler should be pulled in as dependency via `install-scripts.sh`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -78,6 +78,69 @@ jobs:
...
+          docker buildx build --platform ${{ steps.tags.outputs.PLATFORM }} \
+          --tag contour:${{ steps.tags.outputs.ARCH }} \
+          -f .github/ubuntu18/Dockerfile \
+          --load \
+          .
+          docker create --name contour-${{ steps.tags.outputs.ARCH }} contour:${{ steps.tags.outputs.ARCH }}
+          docker cp contour-${{ steps.tags.outputs.ARCH }}:/home/docker/contour/package.deb \
+          "contour-${{ steps.set_vars.outputs.VERSION_STRING }}-ubuntu18.04-amd64.deb"
+          docker cp contour-${{ steps.tags.outputs.ARCH }}:/home/docker/contour/package.ddeb \
+          "contour-dbgsym-${{ steps.set_vars.outputs.VERSION_STRING }}-ubuntu18.04-amd64.ddeb"
```
Comment (christianparpart) [inline .github/workflows/build.yml:127]: Veeery nice. I was afraid that this would be more complicated to get them out of there. :)

**Review by christianparpart — COMMENTED** [review]
Sad thing is, the `release.yaml` needs to be adapted to.

But overall it looks very good. Many many thanks for going through this. :)

File: .github/ubuntu18/Dockerfile (hunk 1 above)
Comment (Yaraslaut) [reply .github/ubuntu18/Dockerfile:34]: Default compiler is g++-8 and to install g++-10 we need to add-apt-repository, i think it's better to have this here than update install-deps.sh script to handle docker image of ubuntu 18.04

Comment (christianparpart) [reply .github/ubuntu18/Dockerfile:34]: Ok. Understood. :)

**Review by christianparpart — APPROVED** [review]
Many thanks! :)

**Pull request merged**